.github/workflows/workflow-dispatch-changeset.yml

```diff
@@ -0,0 +1,40 @@
+on:
+  workflow_dispatch:
+    inputs:
+      generate-changelog:
+        description: 'Pick enter for pre-release or exit for production changeset'
+        required: true
+        default: 'enter'
+        type: choice
+        options:
+          - enter
+          - exit
+
+jobs:
+  generation-changelog-on-trigger:
+    runs-on: ubuntu-latest
+    if: ${{ startsWith( github.ref, refs/heads/v2-release ) }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+      - name: Install Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version-file: '.node-version'
+          cache: 'yarn'
+      - name: Install dependencies
+        run: yarn --immutable
+      - name: Change changeset pre based on trigger
+        run: yarn changeset pre ${{ inputs.generate-changelog }} ${{ inputs.generate-changelog == 'enter' && 'next' }}
+      - name: Create PR with changelog
+        id: changesets
```

Comment:
```suggestion
        id: changesets
        if: ${{ inputs.generate-changelog == 'enter' }}
```

Reply:
it should run always, no matter if its prod or next